pkg/gcp/compute/routes/client.go

```diff
@@ -0,0 +1,78 @@
+package routes
+
+import (
+	"context"
+	"fmt"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"google.golang.org/api/iterator"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+)
+
+type Client struct {
+	routesClient *compute.RoutesClient
+}
+
+func NewClient(client *compute.RoutesClient) *Client {
+	return &Client{
+		routesClient: client,
+	}
+}
+
+func (c *Client) DeleteRoutes(ctx context.Context, cluster *capg.GCPCluster) error {
+	logger := c.getLogger(ctx)
+	project := cluster.Spec.Project
+	filter := fmt.Sprintf(`name : "%s*"`, cluster.Name)
```

Comment:
Maybe it's not possible, but is there a way to limit a bit what will be deleted? I'm worried that if I delete cluster `jose`, it will delete all routes for clusters `jose2` and `jose5`.